### DISCUSSION TITLE
Explore source code for understanding how Longhorn works

### DISCUSSION THREAD

**Discussion by alexey-sderzhikov** [longhorn/longhorn#4553] [General]
Hi! Thanks Longhorn's team and other contributors for this project, its inspiring me!
I want to get deeper in understanding how Longhorn works in background, how writes data between nodes, how creates volumes, how creates backups and etc.

Which repos/packages/files do i need to explore first for a general understanding?

Thanks for answers!

**Comment by derekbit** [comment]
The doc in https://longhorn.io/docs/1.3.1/concepts/ can help you understand the big picture and the components' interactions in LH.

**Reply by alexey-sderzhikov** [reply]
Yeah, i already read concepts and another things in docs, but now i want to explore source code, but don't know where to start

**Reply by derekbit** [reply]
I think we don't have detailed documents for the source codes. You can check the [wiki](https://github.com/longhorn/longhorn/wiki) for getting started with the development and the general information.

In general, the control plane logics are in [longhorn-manager](https://github.com/longhorn/longhorn-manager), and [longhorn-engine ](https://github.com/longhorn/longhorn-engine)is responsible for the data plane. The two repo can be regarded as the entry points.

Additionally, you can read the [enhancement proposals](https://github.com/longhorn/longhorn/tree/master/enhancements) in which developers describe the probs, designs and implementations.
